main.go

```diff
@@ -173,42 +185,41 @@ func main() {
-						log.Printf("PossibleUsername: %v", possibleUsername)
-						log.Printf("Usernames: %v", usernames)
-						if _, ok := usernames[possibleUsername]; ok {
-							log.Printf("Found username, adding to notifications %s!\n", possibleUsername)
-							pinger := c.Users[event.UserID].Nick
-							inOneHour := time.Now().In(location).Add(5 * time.Second)
-                            addNotification(inOneHour, possibleUsername, pinger, event.ThreadID, event.Flow)
-							//notifications[usernames[possibleUsername]][event.ThreadID] = Notification{inOneHour, event.ThreadID, event.Flow, pinger}
-							notifyTag := fmt.Sprintf("notify-short-%v", possibleUsername)
-							flowdock.EditMessageInFlowWithApiKey(flowdockAPIKey, org, flow, strconv.FormatInt(event.ID, 10), "", []string{notifyTag})
-						}
-						log.Printf("%s", notifications[event.UserID][event.ThreadID])
+				var notifRegex = regexp.MustCompile(fmt.Sprintf(`(\%s+)([\wåäö]+)`, prefix))
+
+				for _, field := range notifRegex.FindAllStringSubmatch(event.Content, -1) {
+					if len(field) < 2 {
+						continue
 					}
-					if strings.HasPrefix(field, prefix) && !strings.HasPrefix(field, strings.Repeat(prefix,2)) {
-						log.Printf("Saw notification thingie, %s", field)
-						possibleUsername := strings.TrimRight(strings.ToLower(strings.TrimLeft(field, prefix)), ",.? ")
-						if _, ok := usernames[possibleUsername]; ok {
-							log.Printf("Found username, adding to notifications %s!\n", possibleUsername)
-							pinger := c.Users[event.UserID].Nick
-							nextWorkDayAtNine := NextWorkdayAtNine()
-							notifications[usernames[possibleUsername]][event.ThreadID] = Notification{nextWorkDayAtNine, event.ThreadID, event.Flow, pinger}
-							notifyTag := fmt.Sprintf("notify-long-%v", possibleUsername)
-							flowdock.EditMessageInFlowWithApiKey(flowdockAPIKey, org, flow, strconv.FormatInt(event.ID, 10), "", []string{notifyTag})
-							saveNotifications(notifications, notificationStorage)
-						}
-						log.Printf("%s", notifications[event.UserID][event.ThreadID])
+					possiblePrefix := field[1]
+					possibleUsername := field[2]
+					// Check first if the username is a known username, if not skip
+					if _, ok := usernames[possibleUsername]; !ok {
+						continue
+					}
+					pinger := c.Users[event.UserID].Nick
+
+					var notificationTime time.Time
+					var notifyTag string
+					if possiblePrefix == fastPrefix {
+						notificationTime = time.Now().In(location).Add(fastDelay)
+						notifyTag = fmt.Sprintf("notify-short-%v", possibleUsername)
+					}
+					if possiblePrefix == slowPrefix {
+						notificationTime = NextWorkdayAtNine()
+						notifyTag = fmt.Sprintf("notify-long-%v", possibleUsername)
+					}
+					if possiblePrefix == fasterPrefix {
+						notificationTime = time.Now().In(location).Add(fasterDelay)
+						notifyTag = fmt.Sprintf("notify-shorter-%v", possibleUsername)
 					}
```

Comment:
very good point, the behaviour after was kind of undefined I added a check for that, the your suggestion would also have worked